python/pyspark/sql/dataframe.py

```diff
@@ -482,10 +482,22 @@ def show(self, n=20, truncate=True, vertical=False):
          age  | 5
          name | Bob
         """
+
+        if not isinstance(n, int) or isinstance(n, bool):
+            raise TypeError(f'Parameter `n` (number of rows) must be an int')
+
+        if not isinstance(vertical, bool):
+            raise TypeError(f'Parameter `vertical` must be a bool')
+
         if isinstance(truncate, bool) and truncate:
             print(self._jdf.showString(n, 20, vertical))
         else:
-            print(self._jdf.showString(n, int(truncate), vertical))
+            try:
+                int_truncate = int(truncate)
+                print(self._jdf.showString(n, int_truncate, vertical))
```

Comment:
I think you should pull this call out, and don't combine with parameter validation logic together.